kivy/core/window/_window_sdl2.pyx

```diff
@@ -686,3 +707,36 @@ cdef SDL_Surface* flipVert(SDL_Surface* sfc):
         memcpy(&rpixels[pos], &pixels[(pxlength - pos) - pitch], pitch)
 
     return result
+
+cdef SDL_Surface* surface_from_core_image(core_image):
+    # Create an SDL_Surface from a core image's texture pixels. Use
+    # SDL_FreeSurface(surface) to deallocate.
+    #
+    # Returns NULL if the core image has no texture pixels, or if
+    # there's a problem with the creation of an SDL Surface.
+    cdef int width, height
+    width, height = core_image.size
+    cdef int pitch = width * 4
+    cdef bytes pixels
+    cdef char* sdl_error
```

Comment:
Probably avoid the cdef here,

```python
the_error = <bytes>SDL_GetError()
```

The resulting bytes object is garbage collected by the interpreter, as opposed to the char pointer

Reply:
This one [I've actually looked up](https://wiki.libsdl.org/SDL_GetError) sometime this month. There's no need to memory manage the `char*` returned from `SDL_GetError()`.

However, it isn't great and `<bytes>` isn't great either because I'm doing some string interpolation with its value below, and for py3 it should be decoded instead.

It actually should be decoded either way if you wanted to do proper Unicode handling in the whole Kivy codebase, but that's another can of worms.